spec/globus/client_spec.rb

```diff
@@ -88,45 +83,68 @@
+  describe "public class methods" do
+    let(:fake_instance) do
+      instance_double(
+        described_class,
+        mkdir: nil,
+        file_count: nil,
+        total_size: nil
+      )
     end
-  end
 
-  describe ".file_count" do
-    it "invokes file_count on the Endpoint class, injecting config" do
-      described_class.file_count(user_id:, work_id:, work_version:)
-      expect(fake_endpoint).to have_received(:file_count).once
+    before do
+      allow(described_class).to receive(:instance).and_return(fake_instance)
     end
-  end
 
-  describe ".total_size" do
-    it "invokes total_size on the Endpoint class, injecting config" do
-      described_class.total_size(user_id:, work_id:, work_version:)
-      expect(fake_endpoint).to have_received(:total_size).once
+    describe ".mkdir" do
+      it "invokes mkdir on the Endpoint class, injecting config" do
+        described_class.mkdir
+        expect(fake_instance).to have_received(:mkdir).once
+      end
     end
-  end
 
-  describe "#mkdir" do
-    it "invokes mkdir on the Endpoint class, injecting config" do
-      client.mkdir(user_id:, work_id:, work_version:)
-      expect(fake_endpoint).to have_received(:mkdir).once
+    describe ".file_count" do
+      it "invokes file_count on the Endpoint class, injecting config" do
+        described_class.file_count
+        expect(fake_instance).to have_received(:file_count).once
+      end
     end
-  end
 
-  describe "#file_count" do
-    it "returns the number of files" do
-      client.file_count(user_id:, work_id:, work_version:)
-      expect(fake_endpoint).to have_received(:file_count).once
+    describe ".total_size" do
+      it "invokes total_size on the Endpoint class, injecting config" do
+        described_class.total_size
+        expect(fake_instance).to have_received(:total_size).once
+      end
     end
   end
 
-  describe "#total_size" do
-    it "returns the size of all files for the path" do
-      client.total_size(user_id:, work_id:, work_version:)
-      expect(fake_endpoint).to have_received(:total_size).once
+  describe "public instance methods" do
+    let(:fake_endpoint) { instance_double(described_class::Endpoint, mkdir: nil, set_permissions: nil, file_count: 3, total_size: 3333) }
+
+    before do
+      allow(described_class::Endpoint).to receive(:new).and_return(fake_endpoint)
+    end
+
+    describe "#mkdir" do
+      it "invokes mkdir on the Endpoint class, injecting config" do
+        client.mkdir
+        expect(fake_endpoint).to have_received(:mkdir).once
+      end
+    end
+
+    describe "#file_count" do
+      it "returns the number of files" do
+        client.file_count
+        expect(fake_endpoint).to have_received(:file_count).once
+      end
+    end
+
+    describe "#total_size" do
+      it "returns the size of all files for the path" do
+        client.total_size
+        expect(fake_endpoint).to have_received(:total_size).once
+      end
     end
```

Comment:
This was all as it was before but moved into its own block to show that the fake_endpoint business is only relevant in this context.